charts/sourcegraph/templates/otel-collector/otel-collector.Deployment.yaml

```diff
@@ -0,0 +1,116 @@
+    {{- include "sourcegraph.labels" . | nindent 4 }}
+    {{- if .Values.otelCollector.gateway.labels }}
+      {{- toYaml .Values.otelCollector.gateway.labels | nindent 4 }}
+    {{- end }}
+    deploy: sourcegraph
+    app.kubernetes.io/component: otel-collector
+  name: {{ .Values.otelCollector.gateway.name }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "sourcegraph.selectorLabels" . | nindent 6 }}
+      app: otel-collector
+  minReadySeconds: 5
+  progressDeadlineSeconds: 120
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: otel-collector
+      {{- if .Values.sourcegraph.podAnnotations }}
+      {{- toYaml .Values.sourcegraph.podAnnotations | nindent 8 }}
+      {{- end }}
+      {{- if .Values.otelCollector.gateway.podAnnotations }}
+      {{- toYaml .Values.otelCollector.gateway.podAnnotations | nindent 8 }}
+      {{- end }}
+      labels:
+      {{- include "sourcegraph.selectorLabels" . | nindent 8 }}
+      {{- if .Values.sourcegraph.podLabels }}
+      {{- toYaml .Values.sourcegraph.podLabels | nindent 8 }}
+      {{- end }}
+      {{- if .Values.otelCollector.gateway.podLabels }}
+      {{- toYaml .Values.otelCollector.gateway.podLabels | nindent 8 }}
+      {{- end }}
+        deploy: sourcegraph
+        app: otel-collector
+    spec:
+      containers:
+        - name: otel-collector
+          image: {{ include "sourcegraph.image" (list . "otelCollector") }}
+          imagePullPolicy: {{ .Values.sourcegraph.image.pullPolicy }}
+          command:
+            - "/bin/otelcol-sourcegraph"
+            - "--config=/etc/otel-collector/config.yaml"
+          {{- with .Values.otelCollector.gateway.args }}
+          args:
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
+          terminationMessagePolicy: FallbackToLogsOnError
+          # TODO: set format and backend vars 
+          env:
+          {{- range $name, $item := .Values.otelCollector.gateway.env}}
+          - name: {{ $name }}
+            {{- $item | toYaml | nindent 10 }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.otelCollector.gateway.resources | nindent 10 }}
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 13133
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 13133
+          ports:
+            - containerPort: 55679 # TODO: parametrize if also done in config
+              name: zpages
+            - containerPort: 4317 # TODO: parametrize if also done in config
+              name: otlp-grpc
+            - containerPort: 4318 # TODO: parametrize if also done in config
+              name: otlp-http
+            # TODO: do we need these?
+            #- containerPort: 14250
+            #  name: jaeger-grpc
+            #- containerPort: 14268
+            #  name: jaeger-http
+            #- containerPort: 9411
+            #  name: zipkin
```

Comment:
I guess we only need to open the jaeger ports conditionally if the jaeger agent is enabled